system/Cache/ResponseCache.php

```diff
@@ -99,8 +100,14 @@ public function make($request, ResponseInterface $response): bool
 
         $headers = [];
 
-        foreach ($response->headers() as $header) {
-            $headers[$header->getName()] = $header->getValueLine();
+        foreach ($response->headers() as $name => $value) {
+            if ($value instanceof Header) {
+                $headers[$name] = $value->getValueLine();
+            } else {
+                foreach ($value as $header) {
```

Comment:
If not Header then it's not automatically an array. Why not check if it is an array?